solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -163,6 +155,60 @@ pub mod solana_ibc {
             counterparty_client_id,
         )
     }
+
+    /// Should be called after setting up client, connection and channels.
+    pub fn send_packet<'a, 'info>(
+        ctx: Context<'a, 'a, 'a, 'info, SendPacket<'info>>,
+        port_id: ibc::PortId,
+        channel_id: ibc::ChannelId,
+        data: Vec<u8>,
+        timeout_height: ibc::TimeoutHeight,
+        timeout_timestamp: ibc::Timestamp,
+    ) -> Result<()> {
+        let mut store = storage::from_ctx!(ctx);
+
+        let sequence = store
+            .get_next_sequence_send(&ibc::path::SeqSendPath::new(
+                &port_id,
+                &channel_id,
+            ))
+            .map_err(error::Error::ContextError)
+            .map_err(|err| error!((&err)))?;
+
+        let port_channel_pk = PortChannelPK::try_from(&port_id, &channel_id)
+            .map_err(|e| error::Error::ContextError(e.into()))?;
+
+        let private_store = store.borrow();
+        let port_channel_store = private_store
+            .private
+            .port_channel
+            .get(&port_channel_pk)
+            .ok_or(error::Error::Internal("Port channel not found"))?;
+
+        let channel_end = port_channel_store
+            .channel_end()
+            .map_err(|e| error::Error::ContextError(e.into()))?
+            .ok_or(error::Error::Internal("Channel end doesnt exist"))?;
```

Comment:
```suggestion
        let channel_end = store
            .borrow()
            .private
            .port_channel
            .get(&port_channel_pk)
            .ok_or(error::Error::Internal("Port channel not found"))?
            .channel_end()
            .map_err(|e| error::Error::ContextError(e.into()))?
            .ok_or(error::Error::Internal("Channel end doesnt exist"))?;
```
and then you don’t have to worry about the borrow sticking around.